src/mongoose_client_api/mongoose_client_api_contacts.erl

```diff
@@ -63,14 +63,18 @@ from_json(Req, #{jid := Caller} = State) ->
     CJid = jid:to_binary(Caller),
     Method = cowboy_req:method(Req),
     {ok, Body, Req1} = cowboy_req:read_body(Req),
-    JSONData = jiffy:decode(Body, [return_maps]),
-    Jid = case maps:get(<<"jid">>, JSONData, undefined) of
-              undefined ->
-                  cowboy_req:binding(jid, Req1);
-              J -> J
-          end,
-    Action = maps:get(<<"action">>, JSONData, undefined),
-    handle_request_and_respond(Method, Jid, Action, CJid, Req1, State).
+    case mongoose_client_api:json_to_map(Body) of
+        {ok, JSONData} ->
+            Jid = case maps:get(<<"jid">>, JSONData, undefined) of
+                      undefined -> cowboy_req:binding(jid, Req1);
+                      J when is_binary(J) -> J;
+                      _ -> undefined
+                  end,
+            Action = maps:get(<<"action">>, JSONData, undefined),
+            handle_request_and_respond(Method, Jid, Action, CJid, Req1, State);
+        _ ->
+          mongoose_client_api:bad_request(Req1, State)
```

Comment:
Another place with 2 spaces instead of 4
```suggestion
            mongoose_client_api:bad_request(Req1, State)
```

Reply:
Done. Thanks.